### PR TITLE
[feature]: Ensure Postgres RDS as aws_db_instance has Query Logging enabled, added param group

### PR DESCRIPTION
**Pull request by patrickmukumbu**
# Description

The PR was created to implement the following changes:
- Ensure Postgres RDS as aws_db_instance has Query Logging enabled
- Moved static values to variables.tf in each example
- Updated Makefile to use the latest
- Updated github workflow files
- Added parameter_group resource and specified resource to be used if `parameter_group_name` is not provided

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-rds/blob/feature/enable-query-logging/CHANGELOG.md#113---2023-04-26)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] First PR? Have you deleted `## How to use this template...` from the root `.README.md`?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

**Comment by j0hnmwangi** [comment]
Hi @patrickmukumbu since this is a feature release please edit the PR title to capture that, also add some more details in the  main comment section of the changes done in this PR.

File: examples/complete/variables.tf [hunk 1]
```diff
@@ -1 +1,132 @@
...
+variable "security_group_ingress_config" {
+  type        = any
+  description = "Incoming traffic configuration for the rds security group"
+  default = [
+    {
+      description = "inbound rds traffic"
+      from_port   = 3306
+      to_port     = 3306
+      protocol    = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
```
Comment (j0hnmwangi) [inline examples/complete/variables.tf]: Can we restrict traffic only from the VPC CIDR block atleast? As a security best practice.

File: examples/complete/main.tf [hunk 2]
```diff
@@ -13,51 +13,33 @@ resource "random_password" "rds_pwd" {
...
+  security_group_ingress              = var.security_group_ingress_config
+  security_group_egress               = var.security_group_egress_config
 
   options = {
     option = {
-      option_name = "MARIADB_AUDIT_PLUGIN"
+      option_name = var.option_name
     }
   }
 }
```
Comment (j0hnmwangi) [inline examples/complete/main.tf:63]: Should we also add an example of how parameter group resource is used?

File: variables.tf [hunk 3]
```diff
@@ -351,7 +351,20 @@ variable "create_security_group" {
 
 variable "port" {
   description = "The port on which the DB accepts connections"
+  type        = number
+  default     = 3306
```
Comment (j0hnmwangi) [inline variables.tf:355]: Should we set 3306 as the default? Given the multiple db engines..

File: examples/s3-import/main.tf [hunk 4]
```diff
@@ -14,49 +14,44 @@ resource "random_password" "rds_pwd" {
...
+  engine                = var.engine
+  instance_class        = var.instance_class
+  subnet_ids            = local.database_subnets
+  max_allocated_storage = var.max_allocated_storage
+  name                  = var.db_name
+  username              = random_string.rds_usr.result
+  password              = random_password.rds_pwd.result
 
   s3_import = {
     source_engine_version = "8.0.28"
```
Comment (j0hnmwangi) [inline examples/s3-import/main.tf]: make it a variable

File: examples/s3-import/s3.tf [hunk 5]
```diff
@@ -1,61 +1,37 @@
...
-  bucket        = local.name
-  force_destroy = true
-
+module "s3_bucket_for_mysql_import" {
+  source                 = "boldlink/s3/aws"
+  version                = "2.2.0"
+  bucket                 = var.name
+  sse_sse_algorithm      = "AES256"
+  sse_bucket_key_enabled = false
+  force_destroy          = true
```
Comment (j0hnmwangi) [inline examples/s3-import/s3.tf]: Move values to variables.tf file

File: examples/s3-import/data.tf [hunk 6]
```diff
@@ -22,27 +21,46 @@ data "aws_kms_alias" "rds" {
...
 
+data "aws_iam_policy_document" "assume_policy" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["rds.amazonaws.com"]
```
Comment (j0hnmwangi) [inline examples/s3-import/data.tf]: use datasource to get dns suffix

**Review by j0hnmwangi — CHANGES_REQUESTED** [review]
Please see my comments @patrickmukumbu

File: examples/complete/variables.tf (hunk 1 above)
Comment (patrickmukumbu) [reply examples/complete/variables.tf]: Done

File: examples/complete/main.tf (hunk 2 above)
Comment (patrickmukumbu) [reply examples/complete/main.tf:63]: I have added a note in changelog to show that in a future release @ndegwajohn

File: variables.tf (hunk 3 above)
Comment (patrickmukumbu) [reply variables.tf:355]: The default is 3306, in cases where a different port number is required we provide the port number

File: examples/s3-import/main.tf (hunk 4 above)
Comment (patrickmukumbu) [reply examples/s3-import/main.tf]: Done

File: examples/s3-import/s3.tf (hunk 5 above)
Comment (patrickmukumbu) [reply examples/s3-import/s3.tf]: Done

File: examples/s3-import/data.tf (hunk 6 above)
Comment (patrickmukumbu) [reply examples/s3-import/data.tf]: done

File: examples/minimum/main.tf [hunk 7]
```diff
@@ -13,20 +13,14 @@ resource "random_password" "rds_pwd" {
...
-  #checkov:skip=CKV_AWS_161: "Ensure RDS database has IAM authentication enabled"
-  #checkov:skip=CKV_AWS_157: "Ensure that RDS instances have Multi-AZ enabled"
-
-
   source              = "../../"
   engine              = "mysql"
   vpc_id              = local.vpc_id
   subnet_ids          = local.database_subnets
-  name                = local.db_name
-  deletion_protection = false
```
Comment (j0hnmwangi) [inline examples/minimum/main.tf:26]: can we restore this setting?

**Review by j0hnmwangi — CHANGES_REQUESTED** [review]

File: examples/minimum/main.tf (hunk 7 above)
Comment (patrickmukumbu) [reply examples/minimum/main.tf:26]: Done

**Comment by patrickmukumbu** [comment]
Deleted since I had squashed commits from previous releases. Reopening with unsquashed commits

**Pull request closed without merging**